pulp_ansible/app/tasks/collections.py

```diff
@@ -301,18 +298,24 @@ def import_collection(
     user_facing_logger = logging.getLogger("pulp_ansible.app.tasks.collection.import_collection")
 
     try:
-        with temp_file.file.open() as artifact_file:
-            with tarfile.open(fileobj=artifact_file, mode="r") as tar:
+        with tempfile.NamedTemporaryFile(dir=".") as local_temp_file:
+            with temp_file.file.open() as artifact_file:
+                # If we don't copy the file now, galaxy-importer will attempt to download it.
+                # TODO tell galaxy-importer to work with django-storage file objects
+                shutil.copyfileobj(artifact_file, local_temp_file)
+            local_temp_file.flush()
+            local_temp_file.seek(0)
+            with tarfile.open(fileobj=local_temp_file, mode="r") as tar:
                 manifest_data = json.load(
                     get_file_obj_from_tarball(tar, "MANIFEST.json", temp_file.file.name)
                 )
                 files_data = json.load(
                     get_file_obj_from_tarball(tar, "FILES.json", temp_file.file.name)
                 )
-            url = _get_backend_storage_url(artifact_file)
-            importer_result = process_collection(
-                artifact_file, filename=filename, file_url=url, logger=user_facing_logger
-            )
+            with open(local_temp_file.name) as file_handle:
+                importer_result = process_collection(
+                    file_handle, filename=filename, logger=user_facing_logger
+                )
```

Comment:
This is a hell of a workaround to prevent galaxy-importer to try to be clever.

Reply:
FYI @awcrosby

Reply:
haha. we may have had a reason for this originally, but if we are able to get it working with pulp_ansible passing a file handle to galaxy-importer, then +1 to that